user/views.py

```diff
@@ -1,3 +1,444 @@
+
+class CheckEmail(View): # 이메일 중복확인
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if data['email'] == '':
+                return JsonResponse({'message' : 'NOT_ENTERED_EMAIL'}, status = 400)
+
+            p = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            if p.match(str(data['email'])) == None:
+                return JsonResponse({'message':'EMAIL_VALIDATION'}, status = 400)
+
+            if User.objects.filter(email = data['email']).exists():
+                return JsonResponse({'message' : 'EMAIL_DUPLICATED'}, status = 400)
+            else:
+                return JsonResponse({'message' : 'SUCCESS'}, status = 200)
+
+        except KeyError as ex:
+            return JsonResponse({'message' : 'KEY_ERROR_' + ex.args[0]}, status = 400)
+        except Exception as ex:
+            return JsonResponse({'message' : 'ERROR_' + ex.args[0]}, status = 400)
+
+class SignIn(View): # 로그인
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if User.objects.filter(user_id = data['user_id']).exists():
+                user_data = User.objects.get(user_id = data['user_id'])
+
+                if bcrypt.checkpw(data['password'].encode('utf-8'), user_data.password.encode('utf-8')) == False:
+                    return JsonResponse({'message' : 'INVALID_USER'}, status = 400)
+                
+                access_token = jwt.encode({'user_id' : data['user_id']}, SECRET, algorithm = ALGORITHM)
+
+                user_dic = {}
+                user_dic['id']            = user_data.id
+                user_dic['user_name']     = user_data.user_name
+                user_dic['email']         = user_data.email
+                user_dic['phone']         = user_data.phone
+                user_dic['address']       = user_data.address
+                user_dic['date_of_birth'] = user_data.date_of_birth
+                user_dic['user_rank']     = UserRank.objects.filter(user = user_data.id).get().name
+
+                return JsonResponse({'message' : 'SUCCESS', 'authorization' : access_token.decode('utf-8'), 'user' : user_dic}, status = 200)
+            else:
+                return JsonResponse({'message' : 'INVALID_USER'}, status = 400)                    
+            
+        except KeyError as ex:
+            return JsonResponse({'message' : 'KEY_ERROR_' + ex.args[0]}, status = 400)
+        except Exception as ex:
+            return JsonResponse({'message' : 'ERROR_' + ex.args[0]}, status = 400)
+
+class FindID(View): # 회원 아이디 찾기
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if User.objects.filter(user_name = data['user_name'], email = data['email']).exists():
+                user_id = User.objects.filter(user_name = data['user_name'], email = data['email']).get().user_id
```

Comment:
```suggestion
                user_id = User.objects.get(user_name = data['user_name'], email = data['email']).user_id
```
하나만 특정해서 객체를 가져올 때에는 filter와 get을 같이 쓰지 않으셔도 됩니다.

Reply:
수정했습니다~